server/asset/btc/btc_test.go

```diff
@@ -303,73 +303,77 @@ func txOutID(txHash *chainhash.Hash, index uint32) string {
 	return txHash.String() + ":" + strconv.Itoa(int(index))
 }
 
-const optimalFeeRate uint64 = 24
-
-func (*testNode) EstimateSmartFee(confTarget int64, mode *btcjson.EstimateSmartFeeMode) (*btcjson.EstimateSmartFeeResult, error) {
-	optimalRate := float64(optimalFeeRate) * 1e-5
-	// fmt.Println((float64(optimalFeeRate)*1e-5)-0.00024)
-	return &btcjson.EstimateSmartFeeResult{
-		Blocks:  2,
-		FeeRate: &optimalRate,
-	}, nil
-}
-
-// Part of the btcNode interface.
-func (t *testNode) GetTxOut(txHash *chainhash.Hash, index uint32, _ bool) (*btcjson.GetTxOutResult, error) {
-	testChainMtx.RLock()
-	defer testChainMtx.RUnlock()
-	outID := txOutID(txHash, index)
-	out := testChain.txOuts[outID]
-	// Unfound is not an error for GetTxOut.
-	return out, nil
-}
-
-// Part of the btcNode interface.
-func (t *testNode) GetRawTransactionVerbose(txHash *chainhash.Hash) (*btcjson.TxRawResult, error) {
-	testChainMtx.RLock()
-	defer testChainMtx.RUnlock()
-	tx, found := testChain.txRaws[*txHash]
-	if !found {
-		return nil, fmt.Errorf("test transaction not found\n")
-	}
-	return tx, nil
-}
-
-// Part of the btcNode interface.
-func (t *testNode) GetBlockVerbose(blockHash *chainhash.Hash) (*btcjson.GetBlockVerboseResult, error) {
-	testChainMtx.RLock()
-	defer testChainMtx.RUnlock()
-	block, found := testChain.blocks[*blockHash]
-	if !found {
-		return nil, fmt.Errorf("test block not found")
-	}
-	return block, nil
-}
-
-// Part of the btcNode interface.
-func (t *testNode) GetBlockHash(blockHeight int64) (*chainhash.Hash, error) {
-	testChainMtx.RLock()
-	defer testChainMtx.RUnlock()
-	hash, found := testChain.hashes[blockHeight]
-	if !found {
-		return nil, fmt.Errorf("test hash not found")
-	}
-	return hash, nil
-}
-
-// Part of the btcNode interface.
-func (t *testNode) GetBestBlockHash() (*chainhash.Hash, error) {
-	testChainMtx.RLock()
-	defer testChainMtx.RUnlock()
-	bbHash := testBestBlock.hash
-	return &bbHash, nil
-}
-
-func (t *testNode) RawRequest(string, []json.RawMessage) (json.RawMessage, error) {
-	if t.rawErr != nil {
-		return nil, t.rawErr
+func (t *testNode) Shutdown() {}
+
+func (t *testNode) WaitForShutdown() {}
+
+func (t *testNode) RawRequest(_ context.Context, method string, params []json.RawMessage) (json.RawMessage, error) {
+	switch method {
+	case methodGetBestBlockHash:
+		testChainMtx.RLock()
+		defer testChainMtx.RUnlock()
+		bbHash := testBestBlock.hash
+		return json.Marshal(bbHash.String())
+	case methodGetBlock:
+		testChainMtx.RLock()
+		defer testChainMtx.RUnlock()
+		var blkHash string
+		_ = json.Unmarshal(params[0], &blkHash)
```

Comment:
I suggest either checking this error, or if we never expect an error (probably this), replace with something like `mustUnmarshal(params[0], &blkHash)` where you use a helper:

```go

func mustUnmarshal(data []byte, thing interface{}) {
	if err := json.Unmarshal(data, thing); err != nil {
		panic(err.Error())
	}
}
```

Reply:
💯